Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/DocumentPropertyEditor.cpp

```diff
@@ -383,6 +389,11 @@ namespace AzToolsFramework
         : QFrame(parentWidget)
     {
         m_layout = new QVBoxLayout(this);
+
+        m_handlerCleanupTimer = new QTimer(this);
+        m_handlerCleanupTimer->setSingleShot(true);
+        m_handlerCleanupTimer->setInterval(0);
+        connect(m_handlerCleanupTimer, &QTimer::timeout, this, &DocumentPropertyEditor::CleanupReleasedHandlers);
```

Comment:
you can replace this with something like this.

```
QTimer::singleShot(0, this, &DocumentPropertyEditor::CleanupReleasedHandlers);
```

Reply:
Actually those can lead to subtly different behaviors!

Using `QTimer::singleShot` in the manner you prescribed will lead to multiple cleanup calls queued for the next tick, which is a mild pessimization unless it's gated by something like a `bool m_cleanupHandlersQueued` to avoid doing so, at which point you'll probably want to introduce a tiny helper method to wrap this whole thing... which is fine, but we avoid it outright by just using a member `QTimer` instead and relying on `start` being a "start or restart" action. I would personally advocated for using this pattern in most debouncing scenarios.